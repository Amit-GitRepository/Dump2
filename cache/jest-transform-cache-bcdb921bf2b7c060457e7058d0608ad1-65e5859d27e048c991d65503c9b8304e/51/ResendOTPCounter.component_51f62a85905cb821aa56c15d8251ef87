Object.defineProperty(exports,"__esModule",{value:true});var cov_hvftwa5fe=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/OTPPrompt/ResendOTPCounter.component.js',hash='5b4990304a5874c71a4c3bc8216ff05dd31a6cba',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/OTPPrompt/ResendOTPCounter.component.js',statementMap:{'0':{start:{line:14,column:4},end:{line:18,column:5}},'1':{start:{line:15,column:6},end:{line:15,column:27}},'2':{start:{line:16,column:6},end:{line:16,column:48}},'3':{start:{line:17,column:6},end:{line:17,column:24}},'4':{start:{line:22,column:22},end:{line:22,column:32}},'5':{start:{line:23,column:4},end:{line:29,column:5}},'6':{start:{line:24,column:6},end:{line:26,column:9}},'7':{start:{line:28,column:6},end:{line:28,column:35}},'8':{start:{line:33,column:20},end:{line:33,column:24}},'9':{start:{line:34,column:4},end:{line:36,column:13}},'10':{start:{line:35,column:6},end:{line:35,column:32}},'11':{start:{line:40,column:4},end:{line:40,column:22}},'12':{start:{line:44,column:4},end:{line:44,column:33}},'13':{start:{line:48,column:4},end:{line:52,column:15}},'14':{start:{line:56,column:0},end:{line:59,column:2}},'15':{start:{line:61,column:0},end:{line:64,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:13,column:14},end:{line:13,column:15}},loc:{start:{line:13,column:20},end:{line:19,column:3}},line:13},'1':{name:'(anonymous_1)',decl:{start:{line:21,column:18},end:{line:21,column:19}},loc:{start:{line:21,column:24},end:{line:30,column:3}},line:21},'2':{name:'(anonymous_2)',decl:{start:{line:32,column:15},end:{line:32,column:16}},loc:{start:{line:32,column:21},end:{line:37,column:3}},line:32},'3':{name:'(anonymous_3)',decl:{start:{line:34,column:32},end:{line:34,column:33}},loc:{start:{line:34,column:44},end:{line:36,column:5}},line:34},'4':{name:'(anonymous_4)',decl:{start:{line:39,column:2},end:{line:39,column:3}},loc:{start:{line:39,column:24},end:{line:41,column:3}},line:39},'5':{name:'(anonymous_5)',decl:{start:{line:43,column:2},end:{line:43,column:3}},loc:{start:{line:43,column:26},end:{line:45,column:3}},line:43},'6':{name:'(anonymous_6)',decl:{start:{line:47,column:2},end:{line:47,column:3}},loc:{start:{line:47,column:12},end:{line:53,column:3}},line:47}},branchMap:{'0':{loc:{start:{line:14,column:4},end:{line:18,column:5}},type:'if',locations:[{start:{line:14,column:4},end:{line:18,column:5}},{start:{line:14,column:4},end:{line:18,column:5}}],line:14},'1':{loc:{start:{line:23,column:4},end:{line:29,column:5}},type:'if',locations:[{start:{line:23,column:4},end:{line:29,column:5}},{start:{line:23,column:4},end:{line:29,column:5}}],line:23},'2':{loc:{start:{line:49,column:6},end:{line:51,column:33}},type:'cond-expr',locations:[{start:{line:50,column:10},end:{line:50,column:65}},{start:{line:51,column:10},end:{line:51,column:33}}],line:49}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0,'11':0,'12':0,'13':0,'14':0,'15':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _Shared=require('../../Shared');
var _helper=require('../../../language/i18n/helper');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

ResendOTPCounter=function(_Component){_inherits(ResendOTPCounter,_Component);function ResendOTPCounter(){var _ref;var _temp,_this,_ret;_classCallCheck(this,ResendOTPCounter);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=ResendOTPCounter.__proto__||Object.getPrototypeOf(ResendOTPCounter)).call.apply(_ref,[this].concat(args))),_this),_this.

state={
counter:_this.props.time},_this.


resendOTP=function(){cov_hvftwa5fe.f[0]++;cov_hvftwa5fe.s[0]++;
if(!_this.state.counter){cov_hvftwa5fe.b[0][0]++;cov_hvftwa5fe.s[1]++;
_this.props.sendOTP();cov_hvftwa5fe.s[2]++;
_this.setState({counter:_this.props.time});cov_hvftwa5fe.s[3]++;
_this.startTimer();
}else{cov_hvftwa5fe.b[0][1]++;}
},_this.

updateCounter=function(){cov_hvftwa5fe.f[1]++;var _ref2=(cov_hvftwa5fe.s[4]++,
_this.state),counter=_ref2.counter;cov_hvftwa5fe.s[5]++;
if(counter>0){cov_hvftwa5fe.b[1][0]++;cov_hvftwa5fe.s[6]++;
_this.setState({
counter:counter-1});

}else{cov_hvftwa5fe.b[1][1]++;cov_hvftwa5fe.s[7]++;
clearInterval(_this.interval);
}
},_this.

startTimer=function(){cov_hvftwa5fe.f[2]++;
var superThis=(cov_hvftwa5fe.s[8]++,_this);cov_hvftwa5fe.s[9]++;
_this.interval=setInterval(function(){cov_hvftwa5fe.f[3]++;cov_hvftwa5fe.s[10]++;
superThis.updateCounter();
},1000);
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(ResendOTPCounter,[{key:'componentWillMount',value:function componentWillMount()

{cov_hvftwa5fe.f[4]++;cov_hvftwa5fe.s[11]++;
this.startTimer();
}},{key:'componentWillUnmount',value:function componentWillUnmount()

{cov_hvftwa5fe.f[5]++;cov_hvftwa5fe.s[12]++;
clearInterval(this.interval);
}},{key:'render',value:function render()

{cov_hvftwa5fe.f[6]++;cov_hvftwa5fe.s[13]++;
return _react2.default.createElement(_Shared.ISText,{type:'SEMIBOLD',onPress:this.resendOTP},
this.state.counter?(cov_hvftwa5fe.b[2][0]++,
(0,_helper.translate)('OTP_RESEND_MSG',{time:this.state.counter})):(cov_hvftwa5fe.b[2][1]++,
(0,_helper.translate)('OTP_RESEND')));

}}]);return ResendOTPCounter;}(_react.Component);cov_hvftwa5fe.s[14]++;


ResendOTPCounter.defaultProps={
time:0,
sendOTP:_noop2.default};cov_hvftwa5fe.s[15]++;


ResendOTPCounter.propTypes={
time:_propTypes2.default.number,
sendOTP:_propTypes2.default.func};exports.default=


ResendOTPCounter;