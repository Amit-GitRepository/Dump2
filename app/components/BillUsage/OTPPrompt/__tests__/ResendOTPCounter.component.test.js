import React from 'react';
import renderer from 'react-test-renderer';
import ResendOTPCounter from '../ResendOTPCounter.component'; // Enzyme adapter for React 16

describe('ResendOTPCounter component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ResendOTPCounter />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
