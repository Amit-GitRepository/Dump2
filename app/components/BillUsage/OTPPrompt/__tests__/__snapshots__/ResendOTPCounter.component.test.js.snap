// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResendOTPCounter component renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "color": "#303C46",
      "fontFamily": "SukhumvitSet-SemiBold",
    }
  }
>
  Resend
</Text>
`;
